resip/stack/TransportSelector.hxx

```diff
@@ -181,16 +189,27 @@ class TransportSelector
                   }
                   else if(mTuple.getType() == rhs.mTuple.getType())
                   {
-                     return mTuple.ipVersion() < rhs.mTuple.ipVersion();
+                     if(mTuple.ipVersion() < rhs.mTuple.ipVersion())
+                     {
+                        return true;
+                     }
+                     else if(mTuple.ipVersion() == rhs.mTuple.ipVersion())
+                     {
+                        if (mTuple.getPort() < rhs.mTuple.getPort())
+                        {
+                           return true;
+                        }
+                        else if (mTuple.getPort() == rhs.mTuple.getPort())
+                        {
+                           return mTuple.presentationFormat() < rhs.mTuple.presentationFormat();
```

Comment:
What is the purpose of the presentationFormat comparison?  It does an inet_ntop- it could be quite expensive if it get's down to this check.

Reply:
It is a good question. I am going to analyze a better approach.
The purpose is to find a proper TLS transport in case the service is configured by more than one TLS transport.
For example,
```
transportSelector.addTransport("192.168.1.1", 5060, V4, TLS);
transportSelector.addTransport("192.168.1.2", 5060, V4, TLS);
```

Reply:
The pull request has been updated. The tuple address comparison as `resip::Data` was replaced with the binary one, which is present in `resip::Tuple::operator<()`